beacon_node/eth2-libp2p/src/rpc/handler.rs

```diff
@@ -33,12 +33,40 @@ pub const RESPONSE_TIMEOUT: u64 = 10;
 /// The number of times to retry an outbound upgrade in the case of IO errors.
 const IO_ERROR_RETRIES: u8 = 3;
 
-/// Inbound requests are given a sequential `RequestId` to keep track of. All inbound streams are
-/// identified by their substream ID which is identical to the RPC Id.
-type InboundRequestId = RequestId;
-/// Outbound requests are associated with an id that is given by the application that sent the
-/// request.
-type OutboundRequestId = RequestId;
+/// Identifier of inbound and outbound substreams from the handler's perspective.
+#[derive(Debug, Clone, Copy, Hash, Eq, PartialEq)]
+pub struct SubstreamId(usize);
+
+impl slog::Value for SubstreamId {
```

Comment:
Nice. I've never had the effort to actually implement this. I usually just deference it, this is nicer